apps/desktop/src/lib/pr/prContent.svelte.ts

```diff
@@ -0,0 +1,106 @@
+import { persistedPRBody, persistedPRTitle } from '$lib/config/config';
+import type { DetailedCommit } from '$lib/vbranches/types';
+
+export interface PRContent {
+	title: string;
+	body: string;
+	templateBody: string | undefined;
+}
+
+export interface PRContentParams {
+	isDisplay: boolean;
+	projectId: string;
+	seriesName: string;
+	seriesDescription: string | undefined;
+	existingTitle: string | undefined;
+	existingBody: string | undefined;
+	commits: DetailedCommit[];
+}
+
+export default function getPRContent(params: PRContentParams): PRContent {
+	let templateBody = $state<string | undefined>(undefined);
+	const defaultTitle: string = $derived.by(() => {
+		if (params.isDisplay) return params.existingTitle ?? '';
+
+		// In case of a single commit, use the commit summary for the title
+		if (params.commits.length === 1) {
+			const commit = params.commits[0];
+			return commit?.descriptionTitle ?? '';
+		}
+
+		return params.seriesName;
+	});
+
+	const defaultBody: string = $derived.by(() => {
+		if (params.isDisplay) return params.existingBody ?? '';
+		if (params.seriesDescription) return params.seriesDescription;
+		if (templateBody) return templateBody;
+
+		// In case of a single commit, use the commit description for the body
+		if (params.commits.length === 1) {
+			const commit = params.commits[0];
+			return commit?.descriptionBody ?? '';
+		}
+
+		return '';
+	});
+
+	const persistedBody = persistedPRBody(params.projectId, params.seriesName);
+	const persistedTitle = persistedPRTitle(params.projectId, params.seriesName);
+
+	let inputBody = $state<string | undefined>(undefined);
+	let inputTitle = $state<string | undefined>(undefined);
```

Comment:
```suggestion
	let inputTitle = $state<string>();
```

This will result in the same thing :+1: